src/mpcCoreKit.ts

```diff
@@ -1210,34 +1256,42 @@ export class Web3AuthMPCCoreKit implements ICoreKit, IMPCContext {
       const deviceTSSIndex = TssShareType.DEVICE;
       const factorPub = getPubKeyPoint(factorKey, factorKeyCurve);
 
-      const ec = new EC(this.keyType);
-      const deviceTSSShare = ec.genKeyPair().getPrivate();
       await this.tKey.initialize();
 
       // only consider 2 types of curve, will change the logic in the future if there are more key types curves
-
       // if both keyType library is avaiable, initialize secp256k1 first as secp256k1 is initialize offline
-
       if (this.supportedCurveKeyTypes.has(KeyType.secp256k1)) {
         const importTssBuf = importTssKey.secp256k1 ? Buffer.from(importTssKey.secp256k1, "hex") : undefined;
         // check if key is in the tsslib and keytype exists
         await this.tKey.initializeTss({
           importKey: importTssBuf,
           factorPub,
-          deviceTSSShare,
           deviceTSSIndex,
           tssKeyType: KeyType.secp256k1,
           serverOpts: {
             // selectedServers: [],
             authSignatures: this.state.signatures,
           },
         });
-      }
-      if (this.supportedCurveKeyTypes.has(KeyType.ed25519)) {
+
+        if (this.supportedCurveKeyTypes.has(KeyType.ed25519)) {
+          const importTssBufEd25519 = importTssKey.ed25519 ? Buffer.from(importTssKey.ed25519, "hex") : undefined;
+          // check if key is in the tsslib and keytype exists
+          await this.tKey.initializeTss({
+            importKey: importTssBufEd25519,
+            tssKeyType: KeyType.ed25519,
+            serverOpts: {
+              // selectedServers: [],
+              authSignatures: this.state.signatures,
+            },
+          });
+        }
+      } else {
```

Comment:
hu? why do you duplicate this part from below and nest it here? just move the below part out of the else and condition it like here!?

Reply:
it is different for initialising first curve and 2nd curve.
during initializing 2nd curve, factorPub, and tssIndex are not allowed to be provided,

updating the logic and comment for easier reading